proxy/gateway/java/gateway-remote-module/src/main/java/com/microsoft/azure/gateway/remote/CommunicationEndpoint.java

```diff
@@ -6,16 +6,14 @@
 
 import java.nio.ByteBuffer;
 
-import org.nanomsg.NanoLibrary;
-
 /**
  * An endpoint that is used to communicate with the remote Gateway which uses nanomsg to send and receive messages. 
  *
  */
 class CommunicationEndpoint {
 
     private final String uri;
-    private final NanoLibrary nano;
+    private final NanomsgLibrary nano;
```

Comment:
Should this type sort of *commit* to the `NanomsgLibrary` type? Should the nanomsg stuff be hidden behind another interface?